support/cas-server-support-rest-authentication/src/main/java/org/apereo/cas/adaptors/rest/RestAuthenticationHandler.java

```diff
@@ -77,6 +85,31 @@ protected AuthenticationHandlerExecutionResult authenticateUsernamePasswordInter
         }
         throw new FailedLoginException("Rest endpoint returned an unknown response for " + c.getUsername());
     }
+
+    /**
+     * Resolve {@link MessageDescriptor warnings} from the {@link ResponseEntity authenticationResponse}.
+     *
+     * @param authenticationResponse The response sent by the REST authentication endpoint
+     * @return The warnings for the created {@link AuthenticationHandlerExecutionResult}
+     */
+    protected List<MessageDescriptor> getWarnings(final ResponseEntity<?> authenticationResponse) {
+        List<MessageDescriptor> messageDescriptors = new ArrayList<>();
+
+        var passwordExpirationDate = authenticationResponse.getHeaders().getFirstZonedDateTime("X-Cas-PasswordExpirationDate");
+        if (passwordExpirationDate != null) {
+            var days = Duration.between(Instant.now(), passwordExpirationDate).toDays();
```

Comment:
use `val`.

What Timezone is `Instant.now()` based on?

Reply:
An Instant itself has no timezone.